### DISCUSSION TITLE
Local Files?

### DISCUSSION THREAD

**Discussion by BlameFirewall** [dtcooper/raspotify#581] [Q&A] [answered]
I feel like this must have been asked before but I can't find it anywhere. Is there support for local files? I have a few tracks that are not on spotify that are very dear to me. Is there an option to load up local files on the SD card, or preferably on a USB drive?

If not, can this be a feature request?

Thanks.

**Comment by JasonLG1979** [comment, marked as answer]
No and no.

No you can't play local files with librespot and no this can't be a feature request.

Raspotify is just a packaged version of librespot. This is the wrong place to request librespot features. But as a Librespot dev as far as I know there are no plans to add local playback support. It's been talked about a few times but the consensus has always been that it's not a feature anyone wants to implement for various reasons.

**Reply by BlameFirewall** [reply]
Ah, thanks for the information. Any chance you could summarize or direct me to a conversation about the technical challenges of adding this feature? I'm searching around for "librespot local files" and not finding much. (Or just point me to some better keywords).

Thanks!

**Reply by JasonLG1979** [reply]
It's been brought up in the IRC I'm pretty sure and at least 1 PR. You would need to reverse engineer that part of the API and then implement it in librespot.

My general thought are:

1. No other Spotify Connect devices implement this feature. It's only available in official clients.
2. Librespot has no UI to use to import files.
3. Librespot has no canonical place to store said files on headless systems.
4. Giving Librespot unfettered access to mounted devices is not a good idea and runs afoul to basic security so an "auto-import" feature is off the table.

Even if it is technically possible somehow you'd probably need to do the work yourself.
